src/v/model/record_batch_reader.h

```diff
@@ -47,11 +47,6 @@ concept ReferenceBatchReaderConsumer = requires(
 class record_batch_reader final {
 public:
     using data_t = chunked_circular_buffer<model::record_batch>;
-    struct foreign_data_t {
-        ss::foreign_ptr<std::unique_ptr<data_t>> buffer;
-        size_t index{0};
-    };
```

Comment:
are there details some where about why this is no longer needed?

Reply:
`foreign_data_t` is no longer needed in general as sharing/reading an `iobuf` was made thread-safe in https://github.com/redpanda-data/seastar/pull/149